packages/nx/src/tasks-runner/run-command.ts

```diff
@@ -435,10 +435,26 @@ export function getRunner(
 
   const modulePath: string = getTasksRunnerPath(runner, nxJson);
 
-  let tasksRunner = require(modulePath);
-  // to support both babel and ts formats
-  if (tasksRunner.default) {
-    tasksRunner = tasksRunner.default;
+  let tasksRunner: TasksRunner;
+  try {
+    const maybeTasksRunner = require(modulePath) as
+      | TasksRunner
+      | { default: TasksRunner };
+    // to support both babel and ts formats
+    tasksRunner =
+      'default' in maybeTasksRunner
+        ? maybeTasksRunner.default
+        : maybeTasksRunner;
+  } catch (e) {
+    if (
+      e.code === 'MODULE_NOT_FOUND' &&
+      (modulePath === 'nx-cloud' || modulePath === '@nrwl/nx-cloud')
+    ) {
+      tasksRunner =
+        require('../nx-cloud/nx-cloud-tasks-runner-shell').nxCloudTasksRunnerShell;
+    } else {
+      throw new Error(`Could not find runner configuration for ${runner}`);
+    }
```

Comment:
nit: lets pull this out into a function, maybe something like `getTasksRunnerImpl`